### PR TITLE
Add non-RNG aten ops to aten dialect.

### PR DESCRIPTION
**Pull request by gprateek93**
This commit adds the aten ops which do not require random number
support to aten dialect. This commit also adds some of the missing
torch types.

Signed-Off-By: Prateek Gupta <prateek@nod-labs.com>

### CODE REVIEW

**Review by cathyzhyi — COMMENTED** [review]
The changes looks good overall. I am curious why do we need to add all those ops all together? Are they required to import a certain model? What does `core` aten ops mean here exactly?

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 1]
```diff
@@ -222,14 +222,19 @@ def get_by_triple(self, key: Tuple[str, str, str]):
     "Tensor?[]": "AnyTorchOptionalTensorListType",
     "Tensor[]": "AnyTorchTensorListType",
     "Scalar": "AnyTorchScalarType",
+    "Scalar[]": "AnyTorchScalarListType",
     "Scalar?": "AnyTorchOptionalScalarType",
     "int": "Torch_IntType",
     "int[]": "TorchIntListType",
+    "int[]?": "TorchOptionalIntListType",
```
[the remark this answers is not in the document]
Comment (gprateek93) [reply python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py:229]: This sounds fine to me. I will make the changes. Thanks!

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 2]
```diff
@@ -666,8 +673,746 @@ def emit_with_mutating_variants(key, **kwargs):
...
+        emit("aten::one_hot : (Tensor, int) -> (Tensor)")
+        emit("aten::tile : (Tensor, int[]) -> (Tensor)")
+        emit("aten::sum_to_size : (Tensor, int[]) -> (Tensor)")
+        emit("aten::istft : (Tensor, int, int?, int?, Tensor?, bool, bool, bool?, int?, bool) -> (Tensor)")
+        emit("aten::dstack : (Tensor[]) -> (Tensor)")
+        emit("aten::vstack : (Tensor[]) -> (Tensor)")
+        emit("aten::hstack : (Tensor[]) -> (Tensor)")
+        emit("aten::smm : (Tensor, Tensor) -> (Tensor)")
+        emit("aten::relu6 : (Tensor) -> (Tensor)")
+        # emit("aten::rrelu : (Tensor, Scalar, Scalar, bool, Generator?) -> (Tensor)")
```
[the remark this answers is not in the document]
Comment (gprateek93) [reply python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py:802]: Sure.

**Comment by gprateek93** [comment]
> The changes looks good overall. I am curious why do we need to add all those ops all together? Are they required to import a certain model? What does `core` aten ops mean here exactly?

These are not for a certain model. The current target is to lower all the PyTorch ops mentioned in https://docs.google.com/spreadsheets/d/1Sp4HUjxwMifS5oDQg0yvjqk7hKOpCfKO4jWH4MTGP-k/edit#gid=0 to `aten` dialect.

**Comment by cathyzhyi** [comment]
> > The changes looks good overall. I am curious why do we need to add all those ops all together? Are they required to import a certain model? What does `core` aten ops mean here exactly?
> 
> These are not for a certain model. The current target is to lower all the PyTorch ops mentioned in https://docs.google.com/spreadsheets/d/1Sp4HUjxwMifS5oDQg0yvjqk7hKOpCfKO4jWH4MTGP-k/edit#gid=0 to `aten` dialect.

I see. Thanks for sharing the background!

**Comment by gprateek93** [comment]
@cathyzhyi I have made changes in the naming as you suggested. Please re-review. Thanks!

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 3]
```diff
@@ -302,6 +309,11 @@ def raw_emit_op(operator: JitOperator, f: TextIO, *, traits: List[str],
     multiple_results = len(operator.returns) > 1
     generic_result_name = lambda i: "result" + (str(i) if multiple_results else "")
 
+    # Generate unique operand and result names by replacing common result name with nameless results
+    for return_name in operator.returns:
+        if return_name['name'] in list(operand['name'] for operand in operator.arguments):
+            return_name['name'] += "Res"
```
Comment (cathyzhyi) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py]: nit: "res_" to follow the naming conventions.

**Review by cathyzhyi — APPROVED** [review]
LGTM other than the nit.

**Comment by silvasean** [comment]
Sorry for the late reply, but can we revert this? I don't see much value in front-loading all these ops. We have `torch.operator "aten.blah"` to represent them. There's no specific additional semantics added by this patch, so we might as well keep them opaque as `torch.operator`.

**Comment by silvasean** [comment]
I'm also worried that this gives a false sense of completeness -- the import process is designed to be 1:1 so it's not surprising that so much works already.

**Comment by cathyzhyi** [comment]
yea, this is revert for now in https://github.com/llvm/torch-mlir/pull/529.

**Pull request merged**